inngest/src/main/kotlin/com/inngest/State.kt

```diff
@@ -20,6 +23,19 @@ class State(
         return sb.toString()
     }
 
+    private fun findNextAvailableStepId(id: String): String {
+        if (id !in stepIdsToSeenCount) {
+            stepIdsToSeenCount[id] = 1
+            return id
+        }
+
+        // use the seen count so far for current step, increment for next time
+        val stepNumber = stepIdsToSeenCount[id]
+        stepIdsToSeenCount[id] = stepIdsToSeenCount.getValue(id) + 1
+
+        return "$id:$stepNumber"
```

Comment:
We may also need a check here to see is this step ID already exists.

A user could do something like:
- Run `"my-step"`
- Run `"my-step:1"` (user explicitly specifying this string)
- Run a loop of `"my-step"` steps

I think here this would result in us redeclaring `"my-step:1"` in the first iteration of the loop even though we'd like to skip it and go straight to `"my-step:2"`.

These IDs just being strings means a user can accidentally stumble into our `[ID]:[count]` format and break some stuff. 😄

Reply:
OK pushed a commit to combine both a hash for O(1) in most cases and a loop afterwards just in case a user used that stepId already. Correct me if I'm wrong but is this a bug in the Go SDK then https://github.com/inngest/inngestgo/blob/0a00daba0b2db68ff0f080f787cf63f0a63b44d8/internal/sdkrequest/manager.go#L123-L131 @darwin67 ?

This seems like it could potentially be worth reserving some delimiter characters for metadata if Inngest has other cases where it would want to modify the user provided stepId.

Reply:
I thought of another edge case where the user could reuse a `my-step:n` name after a loop that used it, so I updated the test and logic to handle that too